### DISCUSSION TITLE
What is this blank space when I switch windows?

### DISCUSSION THREAD

**Discussion by rishubn** [paperwm/PaperWM#712] [Q&A] [answered]
See attached video, what is the space labeled: @!0,0BDHF? 
I am using two monitors and I have one of these spaces/windows for each screen, it gets a bit annoying when cycling through windows

[Screencast from 2023-11-25 11:54:44.webm](https://github.com/paperwm/PaperWM/assets/6591138/1844b34a-daf6-4af7-917e-7c374d7b04cc)

**Comment by Lythenas** [comment]
I'm guessing it is a desktop icon extension. Can you check which extensions you have enabled?

**Reply by rishubn** [reply, marked as answer]
You're correct, it was a desktop icon extension. Perhaps the readme could be updated to add a note that the icons have that effect?

**Reply by Lythenas** [reply]
I thought we did, but I can't find it. I created an issue to track this: #713

**Comment by jtaala** [comment]
@rishubn - can you put a link to the desktop icon extension (or any other info there).

It's sounds like that extension has an invisible window - we can generally work around these issues (and keep the extension) by adding the a winprop rule for the invisible window class/title.

If it's a common issue (with this particular) extension we could add it to the default winprop rules.

**Reply by Lythenas** [reply]
How would we work around this with winprops?

The extension just does not work. Ignoring that the position is incorrect (the empty window is supposed to cover the whole desktop), it should show the desktop icons, which it doesn't. We can't fix that.

**Reply by Lythenas** [reply]
If you want to try this yourself. You can e.g. try https://extensions.gnome.org/extension/2087/desktop-icons-ng-ding/

But on gnome 45 this is completely broken with paperwm. Much worse than just showing an empty window.

**Reply by jtaala** [reply]
> It's sounds like that extension has an invisible window...

Well, that comment was based on the assumption above, which sounds like it's not the main issue anyways.

I installed it, and yes, there looks to be much more issues than just invisible windows (also, in PaperWM our backgrounds aren't the actual gnome backgrounds (but specific clutter actors that serve a few purposes) - so, yeah, it just wouldn't work given it looks to make a background with icons.

**Reply by jtaala** [reply]
Is this extension a default extension that is installed and enabled on a distro?

**Reply by Lythenas** [reply]
I'm not sure if it is this extension but there is a desktop icon extension on Ubuntu. I can check tomorrow what it's called. I think it behaves very similarly.

**Reply by rishubn** [reply]
Sorry for the late reply. This is the extension: https://gitlab.com/rastersoft/desktop-icons-ng

it is installed by default on Ubuntu 22.04 LTS

**Comment by jtaala** [comment]
E.g. here's a couple of examples (wl-copy used to do this - create an invisible window):

https://github.com/paperwm/PaperWM/issues/466
